test/Base_Tests/src/Data/Polyglot_Spec.enso

```diff
@@ -42,6 +42,10 @@ add_specs suite_builder = suite_builder.group "Polyglot" group_builder->
         hold.value . should_equal 42
         hold.boxed . should_equal 42
 
+    group_builder.specify "extension method on IntHolder" <|
```

Comment:
Hi guys. I welcome suggestions how to expand the test suite to cover everything _extension methods_ related.

Reply:
The 3 tests already there seem to test the fundamental properties. I like that we also ensure to test the static syntax.

Firstly, I would add some tests that check what happens if there is ambiguity - e.g. if one defines an extension method in Enso that 'shadows' a method with the same name that was already on the Java type.

However, this has never worked right for Enso types (the issue #1243 is 3.5 years old and is still persistent - _checked today_). So I'm not sure if it is in the scope of this PR to get this fixed, but in general I think we probably should get this sorted out.

Reply:
These seem less crucial, but we could also add test cases for following edge cases:
- extension method taking no arguments,
- extension method with default arguments,
	- and in particular - with _all_ arguments defaulted.

(all of these could be done both for static and member method if we want to be thorough)

Knowing that there used to be problems with that, I guess it may also be good to check suspended arguments and suspended+defaulted arguments (related bug #8937). However, if I understand correctly there probably should be no reason why these would work differently than for regular Enso methods.

Reply:
Lastly, given the recently implemented #8867, I guess it may be good to also verify if the typecheck works correctly in case of the extensions on Java types.

Of course your static syntax test already covers the 'happy path'.

So I'd 'just' add one more case:
```
Test.expect_panic Type_Error (IntHolder.value_plus 10 12)
```

that checks that we cannot give an Enso `Integer` for the Java `IntHolder` `self`.

Reply:
Thanks for the test. Used in e279793

Reply:
Yeah, #1243 looks serious - I wonder why it doesn't come up during the weekly planning session. CCing @jdunkerley. I've added `triage` label - maybe it helps.